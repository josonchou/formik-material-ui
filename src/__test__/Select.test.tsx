import * as React from 'react';
import { Formik, Field, Form } from 'formik';
import renderer from 'react-test-renderer';
import MenuItem from '@material-ui/core/MenuItem';

import { Select } from '../Select';

test('Select Renders Correctly', () => {
  const component = renderer.create(
    <Formik onSubmit={() => null} initialValues={{ test: '' }}>
      <Form>
        <Field name="test" label="Select" component={Select}>
          <MenuItem value="">
            <em>None</em>
          </MenuItem>
          <MenuItem value={10}>Ten</MenuItem>
          <MenuItem value={20}>Twenty</MenuItem>
          <MenuItem value={30}>Thirty</MenuItem>
        </Field>
      </Form>
    </Formik>
  );

  expect(component.toJSON()).toMatchSnapshot();
});
