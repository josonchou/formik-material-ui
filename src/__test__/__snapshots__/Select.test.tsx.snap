// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Renders Correctly 1`] = `
<form
  onReset={[Function]}
  onSubmit={[Function]}
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline"
    label="Select"
    onClick={[Function]}
  >
    <div
      className="MuiSelect-root"
    >
      <div
        aria-haspopup="true"
        aria-pressed="false"
        className="MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
        id="select-test"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </div>
      <input
        name="test"
        type="hidden"
        value=""
      />
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root MuiSelect-icon"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </div>
  </div>
</form>
`;
